### PR TITLE
Add variable substitution

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Hey, it would be great to finally support the variable substitution in the library, please tell me what you think about the implementation.

### CODE REVIEW

File: dotenv/src/parse.rs [hunk 1]
```diff
@@ -77,51 +92,78 @@ fn parse_value(input: &str) -> Result<String> {
...
-                //then there's \v \f bell hex... etc
-                match c {
-                    '\\' | '"' | '$' => output.push(c),
-                    _ => return Err(Error::LineParse(input.to_owned())),
+        } else if substitution_mode != SubstitutionMode::None {
+            if c.is_alphanumeric() {
+                substitution_name.push(c);
+            } else {
+                match substitution_mode {
+                    SubstitutionMode::None => panic!("Impossible match branch due to enclosing if condition"),
```
Comment (dgriffen) [inline dotenv/src/parse.rs]: nit: I would use the `unreachable` macro here

Comment (SomeoneToIgnore) [reply dotenv/src/parse.rs]: Thanks for the tip, used it

**Comment by SomeoneToIgnore** [comment]
Great, glad that you're fine with the earlier schedule for this one.

**Comment by dgriffen** [comment]
@SomeoneToIgnore Would you mind writing some examples into the README on how to take advantage of the new feature?

**Comment by SomeoneToIgnore** [comment]
Sure, added.

**Pull request merged**